src/parser/ParseNode.mts

```diff
@@ -0,0 +1,2082 @@
+import type { ArrowInfo } from './Scope.mjs';
+
+export interface Position {
+  line: number;
```

Comment:
That's certainly feasible. I was primarily focused on getting the correct intellisense and checks so that I could make progress on parser support for `using`, and less focused on the mutability of the types.

Reply:
Yes, we can add that later. I always wanted a stricter version of TypeScript that things are immutable by default and use something like `writable` to opt-in.

This won't block the PR but if we have it that will be awesome.

Reply:
I've made all nodes readonly by default, and mutable when passed through `Unfinished`.

Reply:
I've also made the various array types readonly as well.